django/thunderstore/repository/api/experimental/serializers.py

```diff
@@ -95,3 +96,132 @@ class Meta:
             "has_nsfw_content",
             "categories",
         )
+
+
+class PackageUploadAuthorNameField(serializers.SlugRelatedField):
+    def __init__(self, *args, **kwargs):
+        kwargs["slug_field"] = "name"
+        super().__init__(*args, **kwargs)
+
+    def get_queryset(self):
+        return UploaderIdentity.objects.filter(
+            members__user=self.context["request"].user,
+        )
+
+
+class PackageUploadCategoriesField(serializers.RelatedField):
```

Comment:
Needed as using the built in primary key related field, subclassing and setting `get_queryset()` seems to just not work

Reply:
Maybe this could use a slug related field as well? The `PackageCategory` model does after all have a `slug` field that's marked unique.